src/components/AuthWrappers/EnforceLoginStatePageWrapper.tsx

```diff
@@ -21,7 +22,7 @@ const Redirect = ({ redirectTo }: EnforceLoginStatePageWrapperProps) => {
     return encodeURIComponent(`${pathname}${search}${hash}`)
   }, [])
 
-  void router.replace(`${redirectTo}?${CALLBACK_URL_KEY}=${redirectUrl}`)
+  void router.replace(appendWithRedirect(redirectTo ?? SIGN_IN, redirectUrl))
```

Comment:
might also need to do some checks to prevent redirection here if the URL is not a relative path. I will build on this PR!